profiler/profile.go

```diff
@@ -231,6 +232,10 @@ func goroutineProfile(cfg *config) (*profile, error) {
 }
 
 func goroutineWaitProfile(cfg *config) (*profile, error) {
+	if n := runtime.NumGoroutine(); n > cfg.maxGoroutinesWait {
+		return nil, fmt.Errorf("%d goroutines exceeds DD_PROFILING_WAIT_PROFILE_MAX_GOROUTINES limit of %d", n, cfg.maxGoroutinesWait)
```

Comment:
For my understanding : what is the consequence of considering this as an error ?

Reply:
Excellent question!

`goroutineWaitProfile` is called via `(*profiler).collect() -> (p *profiler).runProfile()`. The `collect()` function treats any `runProfile` error as a warning that it puts into the log as well as a metrics counter:

See https://github.com/DataDog/dd-trace-go/blob/v1/profiler/profiler.go#L176-L181

Reply:
Might be good to mention what is happening or not happening since the goroutines exceeded the limit. i.e. `"Skipping wait profile. %d goroutines exceeds..."`